x-pack/plugins/security_solution/cypress/integration/url_compatibility.spec.ts

```diff
@@ -12,13 +12,18 @@ import {
   DATE_PICKER_START_DATE_POPOVER_BUTTON,
   DATE_PICKER_END_DATE_POPOVER_BUTTON,
 } from '../screens/date_picker';
+import { cleanKibana } from '../tasks/common';
 
 const ABSOLUTE_DATE = {
   endTime: '2019-08-01T20:33:29.186Z',
   startTime: '2019-08-01T20:03:29.186Z',
 };
 
 describe('URL compatibility', () => {
+  before(() => {
```

Comment:
How are we deciding between `before` and `beforeEach` in these tests? Unless there's a huge performance cost my preference is always for `beforeEach` to minimize shared state between tests.

Reply:
I understand your point, but there are specific scenarios were there is not going to be shared state of the tests that can impact the behaviour and is going to save us execution time.

Reply:
What are the scenarios you're referring to, here? The use of `cleanKibana` in `before` only works _if each test is allowed to complete its teardown_: 

* If a test fails after it's created a rule and `deleteRule` is part of the test itself, then the rule SO will be persisted to the next test
* If a test fails _before_ it's created a rule and `deleteRule` is part of an `after*` hook, `deleteRule` will fail and cause cypress to skip the remaining tests

I personally would rather have a slow test than a flaky one. Performance and reliability are (with the exception of timeouts) mostly independent issues, and reliability is the focus at the moment. IMO we've also got many more options for improving performance/speed of these tests _external to the tests themselves_ (more CI groups, beefier CI machines, etc.) such that I don't think we need to be sacrificing reliability for performance right now.